src/logger/error.hpp

```diff
@@ -0,0 +1,24 @@
+/* Copyright © 2018 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/*
+ * Error types to distinguish Turi Create errors from arbitrary C++ exceptions
+ */
+
+#include <ios>
+
+namespace turi {
+    namespace error {
+        class io_error: public std::ios_base::failure {
```

Comment:
Formatting nits:

I prefer the precedent in other areas of our code, in which the contents of namespaces are not indented. This makes it easier to change the nesting level of a class's name without worrying about the indent level of the class itself.

Most of our code also seems to use an indent level of 2. I don't have strong feelings either way, although it would be nice to be consistent.